protected/humhub/modules/content/widgets/stream/StreamEntryWidget.php

```diff
@@ -106,13 +106,19 @@ private static function renderLegacyWallEntry(ContentActiveRecord $record, $opti
             $jsWidget = $options['jsWidget'];
             unset($options['jsWidget']);
         } else {
-            $jsWidget = $record->getWallEntryWidget()->jsWidget;
+            $wallEntryWidget = $record->getWallEntryWidget();
+            $jsWidget = $wallEntryWidget instanceof StreamEntryWidget ? $wallEntryWidget->jsWidget : null;
+        }
+
+        if ($jsWidget === null) {
+            Yii::error('Model ' . get_class($record) . ' must define $wallEntryClass or set $streamChannel to null!', 'content');
+            return '';
         }
 
         $params = [
-            'content' => $record->getWallOut($options),
+            'content' => $record instanceof ContentActiveRecord ? $record->getWallOut($options) : '',
```

Comment:
Should we really allow non `ContentActiveRecord`  in the stream?

Reply:
Personally speaking, I would have liked my original idea which would have made this so, but @yurabakhtin's suggestions were definitely a better option in regards to the error reported in #6940, the only issue I had with it was that it requires more coding on my end.

Reply:
@luke- I saw in the first commit [here](https://github.com/humhub/humhub/pull/6941/commits/f97d73f609f5967c95d0d32ca3a03637b0ee7d6a#diff-708348c91e9bdcfac2cf80d59dbff60f3b13e95b8da5bdeb13cedc3783eec5bdR122) @ArchBlood wrote a code like `method_exists($record, 'getWallOut')`, so I suggested to replace it with `$record instanceof ContentActiveRecord`.

Yes, you are right, now we index only `ContentActiveRecord`, because only `Content` records are indexed [here](https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/content/commands/SearchController.php#L42-L45):

```php
foreach (Content::find()->each() as $content) {
    (new ContentSearchService($content))->update(false);
}
```

So maybe we could revert the line back to `$record->getWallOut($options)`, because a main issue was only in a code above `$record->getWallEntryWidget()->jsWidget`.

---

As I understand all `ContentActiveRecord`s become searchable automatically even without `Searchable` interface since `v1.16` because you have added the method `getSearchAttributes` [here](https://github.com/humhub/humhub/commit/92ef2926d64e2cc50d07a0577ff4ca5f53ac88b3#diff-5fea8ec3ab977d4e8e1db5cf83b2509d7cdccff1e612ca16efb88c2a37e18793) for the class `ContentActiveRecord`.
So we can/should remove a code `implements Searchable` from all `ContentActiveRecord`s for versions 1.16+.

Then I think we should test how the searching works for `ContentActiveRecord` models without `implements Searchable` and with `$streamChannel = default` and without `$wallEntryClass = WallEntry::class`.
I find only one such this: [rss-importer/models/RssFeed](https://github.com/humhub/rss-importer/blob/main/models/RssFeed.php#L29).

Reply:
@luke- I have fixed RSS Feed in the PR https://github.com/humhub/rss-importer/pull/19.

Reply:
> @luke- I saw in the first commit [here](https://github.com/humhub/humhub/pull/6941/commits/f97d73f609f5967c95d0d32ca3a03637b0ee7d6a#diff-708348c91e9bdcfac2cf80d59dbff60f3b13e95b8da5bdeb13cedc3783eec5bdR122) @ArchBlood wrote a code like `method_exists($record, 'getWallOut')`, so I suggested to replace it with `$record instanceof ContentActiveRecord`.
> 
> Yes, you are right, now we index only `ContentActiveRecord`, because only `Content` records are indexed [here](https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/content/commands/SearchController.php#L42-L45):
> 
> ```
> foreach (Content::find()->each() as $content) {
>     (new ContentSearchService($content))->update(false);
> }
> ```
> 
> So maybe we could revert the line back to `$record->getWallOut($options)`, because a main issue was only in a code above `$record->getWallEntryWidget()->jsWidget`.
> 
> As I understand all `ContentActiveRecord`s become searchable automatically even without `Searchable` interface since `v1.16` because you have added the method `getSearchAttributes` [here](https://github.com/humhub/humhub/commit/92ef2926d64e2cc50d07a0577ff4ca5f53ac88b3#diff-5fea8ec3ab977d4e8e1db5cf83b2509d7cdccff1e612ca16efb88c2a37e18793) for the class `ContentActiveRecord`. So we can/should remove a code `implements Searchable` from all `ContentActiveRecord`s for versions 1.16+.
> 
> Then I think we should test how the searching works for `ContentActiveRecord` models without `implements Searchable` and with `$streamChannel = default` and without `$wallEntryClass = WallEntry::class`. I find only one such this: [rss-importer/models/RssFeed](https://github.com/humhub/rss-importer/blob/main/models/RssFeed.php#L29).

With the current idea it will throw `must define $wallEntryClass or set $streamChannel to null!` which isn't desirable in the long run. I've tested it multiple times and have set `protected $streamChannel = null;` so the logic is off here, I can agree that `Searchable` isn't needed anymore as I've tested without it and it works as intended, but the issue still remains the same when it comes to the new changes as suggested, you **_MUST_** set `$wallEntryClass` this is not avoidable, so in this case setting `$streamChannel` does nothing at all and is useless.